src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverter.Dynamic.Sample.Tests.cs

```diff
@@ -0,0 +1,315 @@
+            obj["MyInt"] = 43;
+            obj["MyDouble"] = 4.3;
+
+            ((IDictionary<string, object>)obj["MyObject"])["MyString"] = "Hello!!";
+
+            // Use Dictionary for new child property
+            var child = new Dictionary<string, object>();
+            child["ChildProp"] = 1;
+            obj["Child"] = child;
+
+            ((IList<object>)obj["MyArray"]).Clear();
+            ((IList<object>)obj["MyArray"]).Add("1");
+            ((IList<object>)obj["MyArray"])[0]="2"; // replace the "1"
+
+            string json = JsonSerializer.Serialize(obj, options);
+            Assert.Equal(ExpectedJson, json);
+        }
+
+        [Fact]
+        public static void DynamicObject_MissingProperty()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            dynamic obj = JsonSerializer.Deserialize<dynamic>("{}", options);
+
+            // We return null here; ExpandoObject throws for missing properties.
+            Assert.Equal(null, obj.NonExistingProperty);
+        }
+
+        [Fact]
+        public static void DynamicObject_CaseSensitivity()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            dynamic obj = JsonSerializer.Deserialize<dynamic>("{\"MyProperty\":42}", options);
+
+            Assert.Equal(42, (int)obj.MyProperty);
+            Assert.Null(obj.myproperty);
+            Assert.Null(obj.MYPROPERTY);
+
+            options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            options.PropertyNameCaseInsensitive = true;
+            obj = JsonSerializer.Deserialize<dynamic>("{\"MyProperty\":42}", options);
+
+            Assert.Equal(42, (int)obj.MyProperty);
+            Assert.Equal(42, (int)obj.myproperty);
+            Assert.Equal(42, (int)obj.MYPROPERTY);
+        }
+
+        [Fact]
+        public static void NamingPoliciesAreNotUsed()
+        {
+            const string Json = "{\"myProperty\":42}";
+
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            options.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+
+            dynamic obj = JsonSerializer.Deserialize<dynamic>(Json, options);
+
+            string json = JsonSerializer.Serialize(obj, options);
+            Assert.Equal(Json, json);
+        }
+
+        [Fact]
+        public static void NullHandling()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+
+            dynamic obj = JsonSerializer.Deserialize<dynamic>("null", options);
+            Assert.Null(obj);
+        }
+
+        [Fact]
+        public static void QuotedNumbers()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            options.NumberHandling = JsonNumberHandling.AllowReadingFromString |
```

Comment:
Is `JsonNumberHandling.WriteAsString` honored as well?

Reply:
Yes, I'll add a test. Thanks